Modules/Devices/src/Devices.Application/Clients/Commands/ChangeClientSecret/Handler.cs

```diff
@@ -0,0 +1,25 @@
+﻿using Backbone.Modules.Devices.Application.Infrastructure.Persistence.Repository;
+using MediatR;
+
+namespace Backbone.Modules.Devices.Application.Clients.Commands.ChangeClientSecret;
+public class Handler : IRequestHandler<ChangeClientSecretCommand, ChangeClientSecretResponse>
+{
+    private readonly IOAuthClientsRepository _oAuthClientsRepository;
+
+    public Handler(IOAuthClientsRepository oAuthClientsRepository)
+    {
+        _oAuthClientsRepository = oAuthClientsRepository;
+    }
+
+    public async Task<ChangeClientSecretResponse> Handle(ChangeClientSecretCommand request, CancellationToken cancellationToken)
+    {
+        var client = await _oAuthClientsRepository.Find(request.ClientId, cancellationToken);
+
+        var clientSecret = string.IsNullOrEmpty(request.NewSecret) ? PasswordGenerator.Generate(30) : request.NewSecret;
+        client.ClientSecret = clientSecret;
+
+        await _oAuthClientsRepository.Update(client, cancellationToken);
+
+        return new ChangeClientSecretResponse(client.ClientId, client.DisplayName, clientSecret);
```

Comment:
Can you pass the whole client instead and let the Response class do the mapping? Otherwise we will probably fuck up some day and pass the information in the wrong order, which won't be recognized by the compiler because all parameters are of type string